src/main/java/utilities/Parser.java

```diff
@@ -0,0 +1,77 @@
+package utilities;
+import java.util.Scanner;
+
+import command.Command;
+import command.ExitCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import command.DeleteCommand;
+import command.TaskCommand;
+import exception.FormatException;
+import exception.NoInputException;
+import task.TaskList;
+
+public class Parser {
+    public static String addHorizontalLinesAndIndentation(String dialog) {
+        StringBuilder res = new StringBuilder("    ____________________________________________________________\n");
+        Scanner sc = new Scanner(dialog);
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            res.append("      ").append(line.trim()).append("\n"); // Ensure consistent indentation
+        }
+        res.append("    ____________________________________________________________");
+        sc.close();
```

Comment:
might be more efficient to persist scanner across method calls by keeping the scanner open as a property?